### ISSUE TITLE
[BUG]: Basic Jest example doesn't work - fails test, doesn't pick up any set HTML passed in

### ISSUE THREAD

**Issue by a-roberts** [IBMa/equal-access#1949]
### Project

accessibility-checker for Node

### Browser

I'm using the jest boilerplate

### Operating system

MacOS, Windows (probably any)

### Description

I expected the example out of the box to fail (missing lang, title) and then when I added the HTML, for it to pass

```
/**
 * @jest-environment jsdom
 */
'use strict';

describe("Hello World Basics", () => {
    test("Actually works", async () => {
        document.body.innerHTML = `
        <html lang="en">
            <head title="foo">
                <div>
                    <img alt="has-alt" src="hello.png" />
                </div>
            </head>
        </html>`
        await expect(document).toBeAccessible("IMG_BASELINE");
    })
});

```

I get this output

```
(base) Adams-MBP:jest adam$ npm t

> @ibma/aat-boilerplate@3.0.0 test
> node_modules/jest/bin/jest.js test/*.test.js

 FAIL  test/basic.test.js
  Hello World Basics
    ✕ Actually works (482 ms)

  ● Hello World Basics › Actually works

    Scan: IMG_BASELINE
    - Message: Page detected as HTML, but does not have a 'lang' attribute
      Level: violation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/html_lang_exists.html#%7B%22message%22%3A%22Page%20detected%20as%20HTML%2C%20but%20does%20not%20have%20a%20'lang'%20attribute%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_3%22%2C%22ruleId%22%3A%22html_lang_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: Verify there is a way to bypass blocks of content that are repeated on multiple Web pages
      Level: potentialviolation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/html_skipnav_exists.html#%7B%22message%22%3A%22Verify%20there%20is%20a%20way%20to%20bypass%20blocks%20of%20content%20that%20are%20repeated%20on%20multiple%20Web%20pages%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22POTENTIAL%22%5D%2C%22reasonId%22%3A%22Potential_1%22%2C%22ruleId%22%3A%22html_skipnav_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: Missing <title> element in <head> element
      Level: violation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/page_title_exists.html#%7B%22message%22%3A%22Missing%20%3Ctitle%3E%20element%20in%20%3Chead%3E%20element%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_2%22%2C%22ruleId%22%3A%22page_title_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)
      Level: violation
      XPath: /html[1]/body[1]
      Snippet: <body>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/skip_main_exists.html#%7B%22message%22%3A%22The%20page%20does%20not%20provide%20a%20way%20to%20quickly%20navigate%20to%20the%20main%20content%20(ARIA%20%5C%22main%5C%22%20landmark%20or%20a%20skip%20link)%22%2C%22snippet%22%3A%22%3Cbody%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22skip_main_exists%22%2C%22msgArgs%22%3A%5B%5D%7D

      14 |             </head>
      15 |         </html>`
    > 16 |         await expect(document).toBeAccessible("IMG_BASELINE");
         |                                ^
      17 |     })
      18 | });
      19 |

      at Object.toBeAccessible (test/basic.test.js:16:32)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
```

I expected this should pick up the set inner HTML, but it doesn't

If provide nothing in, it doesn't say it's found any html at all

### Steps to reproduce

Clone the repo and navigate to https://github.com/IBMa/equal-access/tree/master/accessibility-checker/boilerplates/jest
npm install that, I'm on Node.js 18.16.1

What am I missing please?

If I take the experience entirely "as is" which is

```
/**
 * @jest-environment jsdom
 */
'use strict';

var path = require("path");

// Describe this Suite of testscases, describe is a test Suite and 'it' is a testcase.
describe("Hello World Basics", () => {
    test("Image missing alt w/ Baseline", async () => {
        document.body.innerHTML = "<div><img src='hello.png' /></div>";
        await expect(document).toBeAccessible("IMG_BASELINE");
    });
    test("Image missing alt without Baseline", async () => {
        document.body.innerHTML = "<div><img src='hello.png' /></div>";
        await expect(document).toBeAccessible("IMG_NO_BASELINE");
    });
});
```

then the result is:

```
(base) Adams-MBP:jest adam$ npm t

> @ibma/aat-boilerplate@3.0.0 test
> node_modules/jest/bin/jest.js test/*.test.js

 FAIL  test/basic.test.js
  Hello World Basics
    ✓ Image missing alt w/ Baseline (377 ms)
    ✕ Image missing alt without Baseline (73 ms)

  ● Hello World Basics › Image missing alt without Baseline

    Scan: IMG_NO_BASELINE
    - Message: Page detected as HTML, but does not have a 'lang' attribute
      Level: violation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/html_lang_exists.html#%7B%22message%22%3A%22Page%20detected%20as%20HTML%2C%20but%20does%20not%20have%20a%20'lang'%20attribute%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_3%22%2C%22ruleId%22%3A%22html_lang_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: Verify there is a way to bypass blocks of content that are repeated on multiple Web pages
      Level: potentialviolation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/html_skipnav_exists.html#%7B%22message%22%3A%22Verify%20there%20is%20a%20way%20to%20bypass%20blocks%20of%20content%20that%20are%20repeated%20on%20multiple%20Web%20pages%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22POTENTIAL%22%5D%2C%22reasonId%22%3A%22Potential_1%22%2C%22ruleId%22%3A%22html_skipnav_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: Missing <title> element in <head> element
      Level: violation
      XPath: /html[1]
      Snippet: <html>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/page_title_exists.html#%7B%22message%22%3A%22Missing%20%3Ctitle%3E%20element%20in%20%3Chead%3E%20element%22%2C%22snippet%22%3A%22%3Chtml%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_2%22%2C%22ruleId%22%3A%22page_title_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)
      Level: violation
      XPath: /html[1]/body[1]
      Snippet: <body>
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/skip_main_exists.html#%7B%22message%22%3A%22The%20page%20does%20not%20provide%20a%20way%20to%20quickly%20navigate%20to%20the%20main%20content%20(ARIA%20%5C%22main%5C%22%20landmark%20or%20a%20skip%20link)%22%2C%22snippet%22%3A%22%3Cbody%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22skip_main_exists%22%2C%22msgArgs%22%3A%5B%5D%7D
    - Message: The image has neither an accessible name nor is marked as decorative or redundant
      Level: violation
      XPath: /html[1]/body[1]/div[1]/img[1]
      Snippet: <img src="hello.png">
      Help: https://able.ibm.com/rules/archives/2024.06.17/doc/en-US/img_alt_valid.html#%7B%22message%22%3A%22The%20image%20has%20neither%20an%20accessible%20name%20nor%20is%20marked%20as%20decorative%20or%20redundant%22%2C%22snippet%22%3A%22%3Cimg%20src%3D%5C%22hello.png%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22fail_no_alt%22%2C%22ruleId%22%3A%22img_alt_valid%22%2C%22msgArgs%22%3A%5B%5D%7D

      14 |     test("Image missing alt without Baseline", async () => {
      15 |         document.body.innerHTML = "<div><img src='hello.png' /></div>";
    > 16 |         await expect(document).toBeAccessible("IMG_NO_BASELINE");
         |                                ^
      17 |     });
      18 | });
      19 |

      at Object.toBeAccessible (test/basic.test.js:16:32)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
Time:        1.743 s, estimated 2 s
Ran all test suites matching /test\/basic.test.js/i.
```

so again I add lang into the one without baseline and I get the same result regardless of what I try.

Updated to add - was really keen to get this working so ran this on my Windows box too with a tiny test case.

Using the same jest boilerplates folder:

```
/**
 * @jest-environment jsdom
 */
'use strict';

var path = require("path");

describe("Hello World Basics", () => {
    test("Works", async () => {
        document.body.innerHTML = "<html><div><img src='hello.png' /></div></html>";
        await expect(document).toBeAccessible("IMG_BASELINE");
    });
});
```

```
 PASS  test/basic.test.js
  Hello World Basics
    √ Works (531 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        3.126 s
```

and yet it's without an alt tag, and IMG_BASELINE.json specifies "Image is missing alt attribute" on line 93
